private/buf/cmd/buf/command/curl/curl.go

```diff
@@ -420,18 +420,36 @@ func (f *flags) validate(isSecure bool) error {
 		return fmt.Errorf("grpc protocol cannot be used with plain-text URLs (http) unless --%s flag is set", http2PriorKnowledgeFlagName)
 	}
 
+	if f.Schema != "" && f.Reflect {
+		if f.flagSet.Changed(reflectFlagName) {
+			// explicitly enabled both
+			return fmt.Errorf("cannot specify both --%s and --%s", schemaFlagName, reflectFlagName)
+		}
+		// Reflect just has default value; unset it since we're going to use --schema instead
+		f.Reflect = false
+	}
+	if !f.Reflect && f.Schema == "" {
+		return fmt.Errorf("must specify --%s if --%s is false", schemaFlagName, reflectFlagName)
+	}
+	schemaIsStdin := strings.HasPrefix(f.Schema, "-")
 	if (len(f.ReflectHeaders) > 0 || f.flagSet.Changed(reflectProtocolFlagName)) && !f.Reflect {
 		return fmt.Errorf(
 			"reflection flags (--%s, --%s) should not be used if --%s is false",
 			reflectHeaderFlagName, reflectProtocolFlagName, reflectFlagName)
 	}
-	if _, err := bufcurl.ParseReflectProtocol(f.ReflectProtocol); err != nil {
-		return fmt.Errorf(
-			"--%s value must be one of %s",
-			reflectProtocolFlagName,
-			stringutil.SliceToHumanStringOrQuoted(bufcurl.AllKnownReflectProtocolStrings),
-		)
+	if f.Reflect {
+		if !isSecure && !f.HTTP2PriorKnowledge {
+			return fmt.Errorf("--%s cannot be used with plain-text URLs (http) unless --%s flag is set", reflectFlagName, http2PriorKnowledgeFlagName)
+		}
```

Comment:
This is the actual fix: an additional check so we can issue a good error message, instead of trying and failing to use reflection over HTTP 1.1.

Reply:
Is the original message unclear? The message I used is similar to others in this function -- since gRPC is also not allowed over HTTP 1.1.